src/libponyrt/mem/alloc.c

```diff
@@ -48,7 +49,8 @@ void* ponyint_virt_alloc(size_t bytes)
 
   if(!ok)
   {
-    perror("out of memory: ");
+    fprintf(stderr, "out of memory trying to allocate %zu bytes: %s\n", bytes, strerror(errno));
```

Comment:
I suggest keeping the `perror` line in case `fprintf` allocates memory.
In the case where you're actually out of memory (and not just trying to allocate a stupidly huge amount) we need to make sure we still print the "out of memory" part.

```suggestion
    perror("out of memory: "); // print this first without allocating memory - fprintf might allocate
    fprintf(stderr, "(tried to allocate %zu bytes)\n", bytes);
```